apps/api/src/app/routes/trading/README.md

```diff
@@ -0,0 +1,185 @@
+
+### Steps
+1. **Get Quote**: Use the getQuote endpoint to retrieve the trade details.
+2. **Sign Order**: Connect your wallet and sign the order using EIP-712 typed data.
+3. **Submit Order**: Send the signed order to the order book.
+
+### Code Example
+
+```js
+(async function() {
+  const trader = {
+    account: '0xfb3c7eb936cAA12B5A884d612393969A557d4307',
+    appCode: 'test1',
+    chainId: 11155111
+  }
+  const params = {
+    kind: 'sell',
+    sellToken: '0xfff9976782d46cc05630d1f6ebab18b2324d6b14',
+    buyToken: '0x0625afb445c3b6b7b929342a04a22599fd5dbb59',
+    amount: '100000000000000000'
+  }
+
+  const callApi = (method, body) => fetch('https://bff.cow.fi/trading/' + method, {
+    method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(body)
+  }).then(res => res.json())
+
+  // Get quote
+  const { quoteResponse, orderToSign, orderTypedData, appDataInfo } = await callApi('getQuote', { trader, params })
+  // Connect wallet
+  const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' })
+  // Sign order
+  const signature = await window.ethereum.request({
+    method: 'eth_signTypedData_v4',
+    params: [accounts[0], JSON.stringify(orderTypedData)]
+  })
+  // Send order
+  const orderId = await callApi('postOrder', {
+    trader,
+    quoteId: quoteResponse.id,
+    signature, // Add order typed data signature (EIP-712)
+    orderToSign,
+    appDataInfo
+  })
+
+  console.log('Order Id:', orderId)
+})()
+```
+
+## 3. Smart-Contract Wallet (EIP-1271 Pre-signing)
+
+For smart-contract wallets, orders are signed using the [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) standard. This involves presigning the order and sending a transaction.
+
+### Code Example
+
+```js
+(async function() {
+  const trader = {
+    account: '0xF568A3a2dfFd73C000E8E475B2D335A4A3818EBa',
+    appCode: 'test1',
+    chainId: 11155111
+  }
+  const params = {
+    kind: 'sell',
+    sellToken: '0xfff9976782d46cc05630d1f6ebab18b2324d6b14',
+    buyToken: '0x0625afb445c3b6b7b929342a04a22599fd5dbb59',
+    amount: '100000000000000000'
+  }
+
+  const callApi = (method, body) => fetch('https://bff.cow.fi/trading/' + method, {
+    method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(body)
+  }).then(res => res.json())
+
+  // Get quote
+  const { quoteResponse, orderToSign, appDataInfo } = await callApi('getQuote', { trader, params })
+  // Connect wallet
+  const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' })
+  // Send order
+  const { orderId, preSignTransaction } = await callApi('postOrder', {
+    trader,
+    quoteId: quoteResponse.id,
+    signingScheme: 'presign', // Signal to use pre-signing (smart-contract wallet)
+    orderToSign,
+    appDataInfo
+  })
+
+  // ACTION NEEDED: Send <preSignTransaction> from smart-contract wallet
```

Comment:
Why not adding the code to send the TX onhcain?

Reply:
It very depends on a smart-contract wallet implementation.
With Safe it requires a lot of code here.

Reply:
For Safe you can go to Transaction build and put the tx data there